SYS-2030: use updated Alma API package

ztucker4 opened the pull request:
Implements [SYS-2030](https://uclalibrary.atlassian.net/browse/SYS-2030)

Updates the CLICC Devices application to use the latest version (0.2.0) of the Alma API Client package.

The only place where Alma data is retrieved is in the `retrieve_sets.py` management command. I made a small change to use the new `APIError` class (and added an additional log message at the end of the process). No logical changes were needed.

To test, rebuild the container to get the latest version of the API client, and run the management command: `python manage.py retrieve_sets`. Tests should also still pass (though we didn't have any for this command specifically): `python manage.py test`

Also includes release notes and tag bump for deployment.


[SYS-2030]: https://uclalibrary.atlassian.net/browse/SYS-2030?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

ztucker4 committed:
SYS-2030: use updated Alma API package

ztucker4 committed:
add release notes and tag bump

akohler commented on clicc_devices/management/commands/retrieve_sets.py:
```diff
@@ -55,11 +55,12 @@ def handle(self, *args, **options) -> None:
                 logger.error(
-                    f"Failed to retrieve Alma set with Alma ID {set_obj.alma_set_id}."
+                    f"Failed to retrieve Alma set with Alma ID {set_obj.alma_set_id}: ",
+                    f"{e.error_messages}",
```
This f-string seems to cause problems with `logger`.  I added a fake set via the admin interface, then ran the `retrieve_sets` command, and got this:
```
--- Logging error ---
Traceback (most recent call last):
  File "/home/django/django_app/clicc_devices/management/commands/retrieve_sets.py", line 59, in handle
    alma_set = alma_client.get_set(set_obj.alma_set_id)
  File "/home/django/.local/lib/python3.13/site-packages/alma_api_client/clients/alma_api_client.py", line 593, in get_set
    raise APIError(api_response, "Error retrieving set")
alma_api_client.models.api.APIError: HTTP 400: Error retrieving set

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.13/logging/__init__.py", line 1151, in emit
    msg = self.format(record)
  File "/usr/local/lib/python3.13/logging/__init__.py", line 999, in format
    return fmt.format(record)
           ~~~~~~~~~~^^^^^^^^
  File "/usr/local/lib/python3.13/logging/__init__.py", line 712, in format
    record.message = record.getMessage()
                     ~~~~~~~~~~~~~~~~~^^
  File "/usr/local/lib/python3.13/logging/__init__.py", line 400, in getMessage
    msg = msg % self.args
          ~~~~^~~~~~~~~~~
TypeError: not all arguments converted during string formatting
Call stack:
  File "/home/django/django_app/manage.py", line 22, in <module>
    main()
  File "/home/django/django_app/manage.py", line 18, in main
    execute_from_command_line(sys.argv)
  File "/home/django/.local/lib/python3.13/site-packages/django/core/management/__init__.py", line 442, in execute_from_command_line
    utility.execute()
  File "/home/django/.local/lib/python3.13/site-packages/django/core/management/__init__.py", line 436, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/home/django/.local/lib/python3.13/site-packages/django/core/management/base.py", line 416, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/home/django/.local/lib/python3.13/site-packages/django/core/management/base.py", line 460, in execute
    output = self.handle(*args, **options)
  File "/home/django/django_app/clicc_devices/management/commands/retrieve_sets.py", line 61, in handle
    logger.error(
Message: 'Failed to retrieve Alma set with Alma ID asdoiasdoaisd: '
Arguments: ("['Set not found: Set ID asdoiasdoaisd']",)
```

This seems to be specific to `logger`, as there's no problem using an f-string with a list.  In the Django shell, so same version of python etc.:
```
$ dc exec django python manage.py shell
10 objects imported automatically (use -v 2 for details).

Python 3.13.5 (main, Jul 22 2025, 01:46:07) [GCC 12.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
from alma_api_client import AlmaAPIClient, APIError
c = AlmaAPIClient("OUR_API_KEY")
try:
  r = c.get_set("asdoiasdoaisd")
except APIError as e:
  print(e.error_messages)
  print(f"{e.error_messages}")

['Set not found: Set ID asdoiasdoaisd']
['Set not found: Set ID asdoiasdoaisd']
>>>
```

akohler reviewed (changes requested):
@ztucker4 The changes themselves look fine.  However, `logger` is not happy with `e.error_messages`, apparently choking on conversion of a list to its output format.

Try converting the list to a string explicitly before sending it to `logger`, probably via `join()`.  If that works, add a comment so we remember why... and this may need a revisit to the Alma script as well, since I didn't test that with an error condition.

Thanks --Andy

ztucker4 committed:
fix problem with error logging

ztucker4 commented:
I think I've fixed the logging issue. The problem wasn't about the list of error messages, but that I had passed two strings to the logger in lines 61-64. I fixed this by removing the comma between the two strings (which I had split for line length reasons), and now messages are concatenated and logged appropriately.

akohler reviewed (approved):
Ah, I see - didn't catch the comma / strings issue, but you're right.  Problem is fixed, log cleanly captures the error, code is merged.
